src/ezlogs_cli.ml

```diff
@@ -1,22 +1,94 @@
-let now_fmt fmt () =
-  let { Unix.tm_year; tm_mon; tm_mday; tm_hour; tm_min; tm_sec; _ } =
-    Unix.gmtime @@ Unix.gettimeofday ()
-  in
-  Fmt.pf fmt "%04d-%02d-%02d %02d:%02d:%02d " (tm_year + 1900) (tm_mon + 1)
-    tm_mday tm_hour tm_min tm_sec
+module Line = struct
+  let now_fmt fmt () =
+    let { Unix.tm_year; tm_mon; tm_mday; tm_hour; tm_min; tm_sec; _ } =
+      Unix.gmtime @@ Unix.gettimeofday ()
+    in
+    Fmt.pf fmt "%04d-%02d-%02d %02d:%02d:%02d " (tm_year + 1900) (tm_mon + 1)
+      tm_mday tm_hour tm_min tm_sec
 
-let pp_header =
-  Fmt.suffix Fmt.(const string " ") @@ Fmt.prefix now_fmt Logs_fmt.pp_header
+  let pp_header =
+    Fmt.suffix Fmt.(const string " ") @@ Fmt.prefix now_fmt Logs_fmt.pp_header
 
-let setup_log style_renderer level =
-  Fmt_tty.setup_std_outputs ?style_renderer ();
-  Logs.set_level level;
-  Logs.set_reporter (Logs_fmt.reporter ~pp_header ());
-  ()
+  let setup style_renderer level =
+    Fmt_tty.setup_std_outputs ?style_renderer ();
+    Logs.set_level level;
+    Logs.set_reporter (Logs_fmt.reporter ~pp_header ());
+    ()
 
-(* Command line interface *)
+  let logging =
+    Cmdliner.Term.(const setup $ Fmt_cli.style_renderer () $ Logs_cli.level ())
+end
 
-open Cmdliner
+module Json = struct
+  module Json = Yojson.Basic
 
-let setup_log =
-  Term.(const setup_log $ Fmt_cli.style_renderer () $ Logs_cli.level ())
+  let dict_of_tags (tags : Logs.Tag.set) =
+    Logs.Tag.fold
+      (fun tag dict ->
+        match tag with
+        | V (tag_definition, tag_value) ->
+          let name = Logs.Tag.name tag_definition in
+          let tag_string =
+            Fmt.str "%a" (Logs.Tag.printer tag_definition) tag_value
+          in
+          (name, `String tag_string) :: dict)
+      tags []
+
+  let reporter ppf =
+    let report src level ~over k msgf =
+      let continuation _ =
+        over ();
+        k ()
+      in
+      let as_json _header tags k ppf fmt =
+        Fmt.kstr
+          (fun message ->
+            let tags_dict =
+              match tags with
+              | None -> []
+              | Some set -> dict_of_tags set
+            in
+            let json : Json.t =
+              `Assoc
+                (("level", `String (Logs.level_to_string (Some level)))
+                :: ("source", `String (Logs.Src.name src))
+                :: ("message", `String message)
+                :: tags_dict
```

Comment:
Just to check, does this include a timestamp and other ECS base level fields?
https://www.elastic.co/guide/en/ecs/current/ecs-base.html

Reply:
Very good call re: including ECS base level fields.
- `@timestamp`: we can get this with filebeat.  I wasn't sure if we should duplicate that in the actual log output or not, since structured logs are likely to be captured and processed by a machine.  Not a problem to add it and format appropriately.
- `labels`: We can drop `tags` directly in this if we want, though that doesn't really seem like a perfect fit.  Maybe we ship a pre-defined [tag definition](https://erratique.ch/software/logs/doc/Logs/Tag/index.html#type-def) for labels?
- `message`: This is in
- `tags`: Unfortunate naming conflict with `Logs`'s concept of tags, but oh well.  Similar to labels, we could/should ship a `Logs` tag definition for this.